### PR TITLE
Add Split Button, update Button to Spectrum CSS v3.0, support direct and distant sibling selector conversion

### LINKED ISSUE #568: Add Spectrum Split Button Pattern
### Expected Behaviour
https://opensource.adobe.com/spectrum-css/components/splitbutton/ should be available in the library

### Actual Behaviour
missing

#### The future
image

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
- Add the `<sp-split-button>` pattern
- Update `<sp-buton>` et al to Spectrum CSS v3.0
- Extend `process-spectrum` to convert more direct sibling (`.something .class + .class`) selectors
- Extend `process-specrrum` to convert distant sibling (`.class ~ .class`) selectors
- correct minor style applications in ActionMenu/Card/Dropdown

## Related Issue
fixes #568 
refs #199 

## Motivation and Context
Support more patters.
Support more CSS.

## How Has This Been Tested?
- unit tests
- visual regressions

## Screenshots (if appropriate):
image

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by andrewhatran** [comment]
This is looking really nice! 🎉 

Currently for the `left` variant of the split button, the main button accepts focus before the trigger button.  Since the trigger is on the left side, should we flip the ordering?

tabOrder

File: packages/split-button/src/SplitButton.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                            : MoreIcon({ hidden: true })}
+                    </sp-icon>
+                </button>
+            `,
+        ];
+        if (this.left) {
+            buttons.reverse();
+        }
+        return html`
+            ${buttons}
```
Comment (Westbrook) [inline packages/split-button/src/SplitButton.ts:162]: @andrewhatran I've updated the rendering order here.

File: packages/split-button/src/SplitButton.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    private trigger!: HTMLButtonElement;
+
+    protected listRole = 'menu';
+    protected itemRole = 'menuitem';
+
+    public focus(): void {
+        if (this.disabled) {
+            return;
+        }
+        if (this.left) {
```
Comment (Westbrook) [inline packages/split-button/src/SplitButton.ts:68]: Ensure that focus enters the element correctly.

File: packages/split-button/src/SplitButton.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            target.click();
+        }
+    }
+
+    protected manageShiftTab(): void {
+        this.addEventListener('keydown', (event) => {
+            const target = event.composedPath()[0];
+            if (
+                target &&
+                target === (this.left ? this.trigger : this.focusElement) &&
```
Comment (Westbrook) [inline packages/split-button/src/SplitButton.ts:99]: Ensure shift + tab leaves the element correctly.

**Pull request merged**